### PR TITLE
Elijah fabon 02 api fact routes

### PR DESCRIPTION
**Pull request by Elijah-Fabon**

### CODE REVIEW

File: app/routes/api.js [hunk 1]
```diff
@@ -39,6 +39,47 @@ router.get('/fact', (req, res) => {
...
+
+    if (!content) {
+        res.status(400).json({ error: 'Content field is required' });
+        return;
+    }
+
+    //const success = addFact({ submitter_id, content, note, discovery_date });
+    const success = addFact({ submitter_id, content, discovery_date, note });
+
+    if (success) {
```
Comment (AlexSichitiu) [inline app/routes/api.js:60]: Doesn't addFact(...) return an object with the success property? So you would be looking at success.success to get the boolean value whereas now you're just checking is success is defined or undefined. Let me know if I'm wrong about this.

File: app/routes/api.js [hunk 2]
```diff
@@ -39,6 +39,47 @@ router.get('/fact', (req, res) => {
     }
 });
 
+// API endpoint to add a new fact to the database.
+router.post('/fact', (req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+    const { userId, content, discovery_date, note } = req.body;
+
+    // v no longer fails foreign key constraint
+    const submitter_id = userId || 'zzz3737';
```
Comment (AlexSichitiu) [inline app/routes/api.js]: Could you define a constant for the anonymous user ID? So we can reuse and change it later.

File: app/routes/api.js [hunk 3]
```diff
@@ -39,6 +39,47 @@ router.get('/fact', (req, res) => {
     }
 });
 
+// API endpoint to add a new fact to the database.
+router.post('/fact', (req, res) => {
+    res.setHeader('Content-Type', 'application/json');
```
Comment (AlexSichitiu) [inline app/routes/api.js]: You don't need to set the header to 'app/json' if you use res.json() (http://restify.com/docs/response-api/#json).

**Review by AlexSichitiu — CHANGES_REQUESTED** [review]
Also could you document these API endpoints in swagger.yaml? In this or a later branch.

File: app/routes/api.js (hunk 2 above)
Comment (Elijah-Fabon) [reply app/routes/api.js:48]: should I just define it in the same file?

File: app/routes/api.js (hunk 1 above)
Comment (Elijah-Fabon) [reply app/routes/api.js:60]: updateFact returns an object, addFact just returns a boolean

**Review by AlexSichitiu — APPROVED** [review]

**Review by AlexSichitiu — APPROVED** [review]

**Pull request merged**